grid2op/Environment/MultiMixEnv.py

```diff
@@ -0,0 +1,218 @@
+    """
+    This class represent a single powergrid configuration,
+    backed by multiple enviromnents parameters and chronics
+
+    It implements most of the BaseEnv public interface:
+    so it can be used as a more classic environment.
+
+    """
+    def __init__(self,
+                 envs_dir):
+        GridObjects.__init__(self)
+        RandomObject.__init__(self)
+
+        self.current_env = None
+        self._envs = []
+
+        # Inline import to prevent cyclical import
+        from grid2op.MakeEnv.Make import make
+
+        try:
+            for env_dir in sorted(os.listdir(envs_dir)):
+                env_path = os.path.join(envs_dir, env_dir)            
+                if not os.path.isdir(env_path):
+                    continue
+                env = make(env_path)
+                self._envs.append(env)
+        except Exception as e:
+            err_msg = "MultiMix environment creation failed: {}".format(e)
+            raise EnvError(err_msg)
+
+        if len(self._envs) == 0:
+            err_msg = "MultiMix envs_dir did not contain any valid env"
+            raise EnvError(err_msg)
+            
+        self.current_env = self._envs[0]
+        # Make sure GridObject class attributes are set from first env
+        # Shouldbe fine since the grid is the same for all envs
+        self.__class__ = self.init_grid(self.current_env)
+
+    def __getattr__(self, name):
+        return getattr(self.current_env, name)
+
+    def reset(self):
+        rnd_env_idx = np.random.randint(len(self._envs))
+        self.current_env = self._envs[rnd_env_idx]
+        self.current_env.reset()
+        return self.get_obs()
+
+    def seed(self, seed=None):
+        """
+        Set the seed of this :class:`Environment` for a better control 
+        and to ease reproducible experiments.
+
+        Parameters
+        ----------
+        seed: ``int``
+           The seed to set.
+
+        Returns
+        ---------
+        seed: ``tuple``
+            The seed used to set the prng (pseudo random number generator) 
+            for all environments
+
+        """
+        try:
+            seed = np.array(seed).astype(dt_int)
+        except Exception as e:
+            raise Grid2OpException("Cannot to seed with the seed provided." \
+                                   "Make sure it can be converted to a" \
+                                   "numpy 64 integer.")
+
+        return super().seed(seed)
+
+    def deactivate_forecast(self):
+        for e in self._envs:
+            e.deactivate_forecast()
+
+    def reactivate_forecast(self):
+        for e in self._envs:
+            e.reactivate_forecast()
+
+    def set_thermal_limit(self, thermal_limit):
+        """
+        Set the thermal limit effectively.
+        Will propagate to all underlying environments
+        """
+        for e in self._envs:
+            e.set_thermal_limit(thermal_limit)
+
+    def get_obs(self):
```

Comment:
Not necessary with the overload of __getattr___ i believe

Reply:
True, same goes for get_thermals and step. Removed these.